src/main/java/org/ow2/proactive/workflow_catalog/rest/service/WorkflowRevisionService.java

```diff
@@ -281,6 +272,32 @@ private QueryExpressionContext createJpaQueryExpression(String workflowCatalogQu
         return ResponseEntity.ok(workflowMetadata);
     }
 
+    public List<WorkflowRevision> getWorkflowsRevisions(Long bucketId, List<Long> idList) {
+        findBucket(bucketId);
+        List<WorkflowRevision> workflows = idList.stream()
+                                                 .map(workflowId -> getWorkflowRevision(bucketId,
+                                                                                        workflowId,
+                                                                                        Optional.empty()))
+                                                 .collect(Collectors.toList());
+        return workflows;
+    }
+
+    private WorkflowRevision getWorkflowRevision(Long bucketId, Long workflowId, Optional<Long> revisionId) {
+        WorkflowRevision workflowRevision;
+
+        if (revisionId.isPresent()) {
```

Comment:
this could be refactored to 

`workflowRevision = revisionId.map(revisionid -> workflowRevisionRepository.getWorkflowRevision(bucketId, workflowId, revisionId.get())).orElse(workflowRepository.getMostRecentWorkflowRevision(bucketId, workflowId))`